Content.Client/HealthAnalyzer/UI/HealthAnalyzerWindow.xaml.cs

```diff
@@ -90,13 +90,15 @@ public void Populate(HealthAnalyzerScannedUserMessage msg)
         }
 
         private void DrawDiagnosticGroups(
-            Dictionary<string, FixedPoint2> groups, IReadOnlyDictionary<string, FixedPoint2> damageDict)
+            Dictionary<string, FixedPoint2> sortedGroups, IReadOnlyDictionary<string, FixedPoint2> damageDict)
         {
             HashSet<string> shownTypes = new();
 
             // Show the total damage and type breakdown for each damage group.
-            foreach (var (damageGroupId, damageAmount) in groups.Reverse())
+            foreach (var (damageGroupId, damageAmount) in sortedGroups.Reverse())
             {
+                // groups are sorted, if the damageAmount == 0
+                // else you can end the cycle
                 if (damageAmount == 0)
                     return;
```

Comment:
Dictionaries aren't sorted so you can't just assume it will be so with reverse iteration.